packages/zoe/src/zoeService/instanceAdminStorage.js

```diff
@@ -1,49 +1,284 @@
 import { E } from '@endo/eventual-send';
-import { makeWeakStore } from '@agoric/store';
+import {
+  canBeDurable,
+  makeScalarBigSetStore,
+  provideDurableWeakMapStore,
+  vivifyKindMulti,
+  vivifyFarClassKit,
+  M,
+  provide,
+} from '@agoric/vat-data';
+import { makeZoeSeatAdminFactory } from './zoeSeat.js';
+import { defineDurableHandle } from '../makeHandle.js';
+import { InstanceAdminI, InstanceHandleShape } from '../typeGuards.js';
 
-/**
- *
- */
-export const makeInstanceAdminStorage = () => {
-  /** @type {WeakStore<Instance,InstanceAdmin>} */
-  const instanceToInstanceAdmin = makeWeakStore('instance');
+const { quote: q, details: X } = assert;
 
-  /** @type {import('./utils.js').GetPublicFacet} */
-  const getPublicFacet = async instanceP =>
-    E.when(instanceP, instance =>
-      instanceToInstanceAdmin.get(instance).getPublicFacet(),
-    );
+const InstanceAdminStorageShape = M.interface('InstanceAdminStorage', {
+  getPublicFacet: M.call(M.eref(InstanceHandleShape)).returns(M.promise()),
+  getBrands: M.call(InstanceHandleShape).returns(M.promise()),
+  getIssuers: M.call(InstanceHandleShape).returns(M.promise()),
+  getTerms: M.call(InstanceHandleShape).returns(M.promise()),
+  getOfferFilter: M.call(M.eref(InstanceHandleShape)).returns(M.promise()),
+  setOfferFilter: M.call(M.eref(InstanceHandleShape)).returns(),
+  getInstallationForInstance: M.call(InstanceHandleShape).returns(M.promise()),
+  getInstanceAdmin: M.call(InstanceHandleShape).returns(InstanceAdminI),
+  initInstanceAdmin: M.call(InstanceHandleShape).returns(M.promise()),
+  deleteInstanceAdmin: M.call(InstanceHandleShape).returns(M.promise()),
+});
 
-  /** @type {GetBrands} */
-  const getBrands = async instance =>
-    instanceToInstanceAdmin.get(instance).getBrands();
+/** @param {import('@agoric/vat-data').Baggage} baggage */
+export const makeInstanceAdminStorage = baggage => {
+  const makeIAS = vivifyFarClassKit(
+    baggage,
+    'InstanceAdmin',
+    InstanceAdminStorageShape,
+    () => ({
+      instanceToInstanceAdmin: provideDurableWeakMapStore(
+        baggage,
+        'instanceToInstanceAdmin',
+      ),
+    }),
+    {
+      // ZoeStorageManager uses the accessor facet to get info about instances
+      accessor: {
```

Comment:
There's something very wrong here, possibly with my portion of the implementation of `define*ClassKit`. You're defining a kit. Your behavior section is correct for a kit. The `accessor` facet looks like a correct implementation of `InstanceAdminStorageShape` which is declared as in interface.

What confuses me is like 35 where you're providing it as the guard argument of a kit. The guard of a kit should be a kit of interface guards of the same shape, not a single interface guard. The fact that it works in its current form is very very bad.

Reply:
I wait until you confirm that I'm not missing something obvious, and then I'll file a bug assigned to me.

In any case, this argument should be a kit of interface guards.

Reply:
Filed as https://github.com/Agoric/agoric-sdk/issues/6656

Reply:
Fixed